File: Coursera Courses/Specializations/Programming in Java: A Hands-on Introduction/Java Object Basics: Functions, Recursion, and Objects/Week 4/LabChallenge.java
import java.util.ArrayList;

//add class definitions below this line

class Zoo {
  int bigCats;
  int primates;
  int reptiles;
  int birds;
  
  Zoo(int bc, int p, int r, int b) {
    bigCats = bc;
    primates = p;
    reptiles = r;
    birds = b;
  }
  
  int totalAnimals() {
    return bigCats+primates+reptiles+birds;
  }
  
  int totalMammals() {
    return bigCats+primates;
  }
  
  String mostAnimals() {
    int max = bigCats;
    String name = "bigCats";
    
    if (primates > max) {
      max = primates;
      name = "primates";
    }
    if (reptiles > max) {
      max = reptiles;
      name = "reptiles";
    }
    if (birds > max) {
      max = birds;
      name = "birds";
    }
    return name;
  }
}

//add class definitions above this line

public class LabChallenge {  
  public static void main(String[] args) {
    
    //add code below this line
    
    Zoo myZoo = new Zoo(10, 30, 90, 120);
    System.out.println(myZoo.totalAnimals());
    System.out.println(myZoo.totalMammals());
    System.out.println(myZoo.mostAnimals());

    //add code above this line
  }
}

File: Coursera Courses/Specializations/Programming in Python: A Hands-on Introduction/Python Object Basics: Functions, Recursion, and Objects/Week 4/lab_challenge.py
class Zoo:
  def __init__(self, big_cats, primates, reptiles, birds):
    self.big_cats = big_cats
    self.primates = primates
    self.reptiles = reptiles
    self.birds = birds
    
  def total_animals(self):
    return self.big_cats + self.primates + self.reptiles + self.birds
  
  def total_mammals(self):
    return self.big_cats + self.primates
  
  def most_animals(self): 
    max = 0
    maxname = ""
    for anim in [self.big_cats, self.primates, self.reptiles, self.birds]:
      if anim > max:
        max = anim
        maxname  = namestr(anim)
    return maxname

def namestr(obj, namespace):
    return [name for name in namespace if namespace[name] is obj]
  
  
my_zoo = Zoo(10, 30, 90, 120)
print(my_zoo.total_animals())
print(my_zoo.total_mammals())
print(my_zoo.most_animals())
